pyrokinetics/readers.py

```diff
@@ -102,13 +109,13 @@ def __setitem__(self, key: str, value: Type[BaseReader]):
             except ValueError as e:
                 raise TypeError(str(e))
 
-        def __delitem__(self, key):
-            self.__dict.pop(key)
-
-        def __iter__(self):
-            return iter(self.__dict)
+    # Set BaseReader as a class-level attribute
+    ReaderFactory.BaseReader = BaseReader
 
-        def __len__(self):
-            return len(self.__dict)
+    # Rename class
+    if name is None:
+        name = f"{BaseReader.__name__}Factory"
+    ReaderFactory.__name__ = name
+    ReaderFactory.__qualname__ = name
 
-    return ReaderFactory()
+    return ReaderFactory, ReaderFactory()
```

Comment:
Possibly I've missed something, but do we need the class itself?

Reply:
Perhaps not. I think I made this change because I wanted to ensure it was creating types with the correct names in my unit tests, but ended up not needing it. It makes an already confusing bit of code even harder to understand, so it's probably best to revert it.